packages/clerk-js/src/core/clerk.ts

```diff
@@ -1470,29 +1471,42 @@ export class Clerk implements ClerkInterface {
   public authenticateWithGoogleOneTap = async (
     params: AuthenticateWithGoogleOneTapParams,
   ): Promise<SignInResource | SignUpResource> => {
-    return this.client?.signIn
-      .create({
-        strategy: 'google_one_tap',
-        token: params.token,
-      })
-      .catch(err => {
-        if (isClerkAPIResponseError(err) && err.errors[0].code === 'external_account_not_found') {
-          return this.client?.signUp.create({
-            strategy: 'google_one_tap',
-            token: params.token,
-            legalAccepted: params.legalAccepted,
-          });
-        }
-        throw err;
-      }) as Promise<SignInResource | SignUpResource>;
+    if (__BUILD_ENABLE_RHC__) {
```

Comment:
I think this is a bit unintuitive since now you have to opt-in to the default behavior. Personally it would be easier to follow if we used this as a guard clause like 
```
if (__BUILD_DISABLE_RHC) {
  clerkUnsupportedEnvironmentWarning('Google One Tap');
  return this.client!.signIn; // TODO: Remove not null assertion
}

... standard implementation
```

Reply:
@jacekradko Agreed, I'll update. At some point it wasn't removing the dead code, thus the current approach.

Reply:
@jacekradko Done